custom-remote-user-store-manager/agent/src/main/java/org/wso2/carbon/remote/user/store/agent/util/DatabaseUtils.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.remote.user.store.agent.util;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.tomcat.jdbc.pool.PoolProperties;
+
+import java.util.Properties;
+import javax.sql.DataSource;
+
+import static org.wso2.carbon.remote.user.store.agent.constant.CommonConstants.CONNECTION_PARAMETER_DRIVER_NAME;
+import static org.wso2.carbon.remote.user.store.agent.constant.CommonConstants.CONNECTION_PARAMETER_MAX_ACTIVE;
+import static org.wso2.carbon.remote.user.store.agent.constant.CommonConstants.CONNECTION_PARAMETER_MAX_IDLE;
+import static org.wso2.carbon.remote.user.store.agent.constant.CommonConstants.CONNECTION_PARAMETER_MAX_WAIT;
+import static org.wso2.carbon.remote.user.store.agent.constant.CommonConstants.CONNECTION_PARAMETER_MIN_EVIC_TABLE_IDLE_TIME_MILLIS;
+import static org.wso2.carbon.remote.user.store.agent.constant.CommonConstants.CONNECTION_PARAMETER_MIN_IDLE;
+import static org.wso2.carbon.remote.user.store.agent.constant.CommonConstants.CONNECTION_PARAMETER_PASSWORD;
+import static org.wso2.carbon.remote.user.store.agent.constant.CommonConstants.CONNECTION_PARAMETER_TEST_WHILE_IDLE;
+import static org.wso2.carbon.remote.user.store.agent.constant.CommonConstants.CONNECTION_PARAMETER_TIME_BETWEEN_EVICTION_RUNS_MILLIS;
+import static org.wso2.carbon.remote.user.store.agent.constant.CommonConstants.CONNECTION_PARAMETER_URL;
+import static org.wso2.carbon.remote.user.store.agent.constant.CommonConstants.CONNECTION_PARAMETER_USER_NAME;
+import static org.wso2.carbon.remote.user.store.agent.constant.CommonConstants.CONNECTION_PARAMETER_VALIDATION_INTERVAL;
+import static org.wso2.carbon.remote.user.store.agent.constant.CommonConstants.CONNECTION_PARAMETER_VALIDATION_QUERY;
+
+public class DatabaseUtils {
+
```

Comment:
Add class level comments (Java Doc)?

Reply:
Fixed in 09486ce0c6187db78c1b97e4eaefd890c511141b.